### PR TITLE
[dv] Extend registers testbench

### PR DESCRIPTION
**Pull request by vogelpi**
This PR extends the registers testbench to also support performance counter CSRs.

What is currently missing:
- ~~Generate randomized transactions over the whole range of available CSRs (currently only contiguous ranges supported)~~ see #474 
- ~~Add read-only/WARL registers~~ see below.

### CODE REVIEW

**Comment by vogelpi** [comment]
FYI @tomroberts-lowrisc

**Comment by imphil** [comment]
@tomroberts-lowrisc @vogelpi what's the status on this one? Is it still relevant now that x macros are used to generate the perf counter-related defines?

**Comment by vogelpi** [comment]
I should rebase this one + I wanted to add another class of register to support variable width registers.

**Comment by vogelpi** [comment]
@tomroberts-lowrisc I finally managed to push this further and would be happy to get your feedback.

I created a new WARL type register class and added the performance monitor counters. Mismatches for the clock cycle counter are currently ignored (this is somehow expected behaviour). We could extend the model to track the number of clock cycles passed between accesses but this would require more changes and is maybe a bit overkill.

Having the WARL type also allows to test all other WARL registers. This is not yet done, but I could do it. WDYT?

File: dv/cs_registers/model/base_register.cc [hunk 1]
```diff
@@ -56,7 +56,17 @@ bool BaseRegister::ProcessTransaction(bool *match, RegisterTransaction *trans) {
       read_val = RegisterClear(trans->csr_wdata);
       break;
   }
-  if (read_val != trans->csr_rdata) {
+  if (trans->csr_addr == 0xB00 || trans->csr_addr == 0xB80) {
```
[the remark this answers is not in the document]
Comment (vogelpi) [reply dv/cs_registers/model/base_register.cc]: Good idea. I have changed that.

**Comment by vogelpi** [comment]
Thanks for your feedback @tomroberts-lowrisc !

**Pull request merged**